### PR TITLE
Improve ci

### PR DESCRIPTION
**Pull request by mguid65**
- Modify CI to use default images from github
    - Add windows, linux, and macos
    - Note: 
         - For Ninja, we have to fetch it from an external source
         - For Windows, we have to do some work to setup the msvc dev environment
         - For both of the above, these are accomplished by using public actions from the marketplace
             - This may not be appropriate and so we may have to do something different

### CODE REVIEW

File: .github/workflows/cxx.yml [hunk 1]
```diff
@@ -1,83 +1,97 @@
-# Copyright © 2024 Bret Brown
-# SPDX-License-Identifier: MIT
-
-name: Test
+name: Test Multi-Platform
 
 on:
   push:
     branches: [ main ]
-    paths:
```
Comment (wusatosi) [inline .github/workflows/cxx.yml:9]: Please remove:
https://github.com/beman-project/exemplar/blob/d9145b52c9072a569d747547edfc7e03a1494809/.github/workflows/ci_tests.yml#L9-L10

If you intend to merge this change.

**Comment by wusatosi** [comment]
Is there any update to this pr?
I wish to propose similar changes like this, is there a specific reason there's no update here?

@neatudarius @camio @bbrown105

**Comment by mguid65** [comment]
I apologize, the reason for the lack of updates is that I forgot I made this PR.

We can either continue with proposed changes to this or we can discard it.

File: CMakePresets.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            "output": {
+                "outputOnFailure": true
+            },
+            "execution": {
+                "noTestsAction": "error",
+                "stopOnFailure": true
+            }
+        }
+    ]
+}
```
Comment (neatudarius) [inline CMakePresets.json:105]:
```suggestion
}

```

**Review by neatudarius — COMMENTED** [review]
@mguid65 , would you like to update your branch with latest main before continue with this review? Some changes are not applicable anymore.

e.g. Due to one usage examples, now the project requires C++20 or later.


Would you like to try to re-enable this example on your updated branch https://github.com/beman-project/exemplar/blob/main/examples/CMakeLists.txt#L4?

File: .github/workflows/cxx.yml [hunk 3]
```diff
@@ -1,83 +1,97 @@
-# Copyright © 2024 Bret Brown
-# SPDX-License-Identifier: MIT
```
Comment (neatudarius) [inline .github/workflows/cxx.yml:2]: why do you want to remove these lines. I think we should keep them.

btw, @camio , should we add a mention for YAML files into https://github.com/beman-project/beman/blob/main/docs/beman-standard.md#file-contents?

**Comment by wusatosi** [comment]
> @mguid65 , would you like to update your branch with latest main before continue with this review? Some changes are not applicable anymore.
> 
> e.g. Due to one usage examples, now the project requires C++20 or later.
> 
> 
> Would you like to try to re-enable this example on your updated branch https://github.com/beman-project/exemplar/blob/main/examples/CMakeLists.txt#L4?

I think this PR is abandoned, see: https://github.com/beman-project/exemplar/issues/27#issuecomment-2371527251, and the comment above.

I can fork this PR and pick this up.

**Comment by neatudarius** [comment]
As per previous reply, I closed this PR to avoid more confusion, we can fork it later.

File: .github/workflows/cxx.yml (hunk 3 above)
Comment (camio) [reply .github/workflows/cxx.yml:2]: > btw, @camio , should we add a mention for YAML files into https://github.com/beman-project/beman/blob/main/docs/beman-standard.md#file-contents?

Sounds like a promising idea. Would you be willing to create a PR or issue for it?

Comment (neatudarius) [reply .github/workflows/cxx.yml:2]: Yes. I will do it.

Comment (neatudarius) [reply .github/workflows/cxx.yml:2]: Merged https://github.com/beman-project/beman/pull/44

**Pull request closed without merging**